### PR TITLE
feat: search progress bar

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Introduce keyframes for the search progress bar.
- Whenever the status gets updated, we increment.
- When the `completedAt` has value, the progress should equal to the number of steps.

Preview:

https://github.com/dailydotdev/apps/assets/13744167/644737f5-0cd7-4af3-8dfa-1c1dc386617e

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/components/search/SearchProgressBar.module.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.animatedBar {
+  background-image: linear-gradient(to right, rgba(113, 71, 237, 0), rgb(206, 61, 243));
+  animation: fadeLeftToRight 1.2s linear infinite;
+}
+
+@keyframes fadeLeftToRight {
```
Comment (rebelchris) [inline packages/shared/src/components/search/SearchProgressBar.module.css:6]: Pulse didnt work?

https://tailwindcss.com/docs/animation#pulse

**Review by rebelchris — APPROVED** [review]
The end time change we do seperate?

File: packages/shared/src/components/search/SearchProgressBar.module.css (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/search/SearchProgressBar.module.css:6]: Unfortunately, the pulse goes full width with the animation.

**Comment by sshanzel** [comment]
> The end time change we do seperate?

Currently based on the create and complete date. Let me check with Valeri when it is going to be available.

**Comment by sshanzel** [comment]
@rebelchris after checking the response, it seems they pass the amount of time taken in the `status`, to avoid overwriting, we are not passing the `completed` status in favor of done.

https://github.com/dailydotdev/apps/pull/2082/commits/11002db5a8011e9d3b4b68395771b199e52d2c52#diff-ac3b17ae84d347e206ee61153877f0728d89c34c4e9e989ba3b56bbaeb559badR124

**Pull request merged**